src/OSPSuite.SimModel/include/SimModel/Observer.h

```diff
@@ -36,6 +37,10 @@ class Observer :
 
 	bool IsConstantDuringCalculation();
 
+	bool IsUsedInFormulas() const;
+	void SetIsUsedInFormulas(bool usedInFormulas);
+
+	bool IsUsedInSimulation() const;
```

Comment:
That's a new feature: for each Observer SimModel estimates now, whether observer is effectively used in the simulation or not. Unused Observers are not calculated later on

Reply:
The idea is to remove as many ODE variables as possible from the final ODE systems.
If an observer is not used elsewhere in the simulation and is not set as persistable, there is no need to calculate it. Otherwise ODE variables involved into the calculation of this observer cannot be removed.